controller/src/main/java/io/pravega/controller/store/stream/InMemoryStream.java

```diff
@@ -451,15 +451,15 @@ public void refresh() {
     }
 
     @Override
-    CompletableFuture<Void> updateActiveTx(int epoch, UUID txId, byte[] data) {
+    CompletableFuture<Void> updateActiveTx(int epoch, UUID txId, Data<Integer> data) {
         Preconditions.checkNotNull(data);
 
         CompletableFuture<Void> result = new CompletableFuture<>();
         synchronized (txnsLock) {
             if (!activeTxns.containsKey(txId.toString())) {
                 result.completeExceptionally(StoreException.create(StoreException.Type.DATA_NOT_FOUND, "active txn"));
             } else {
-                activeTxns.compute(txId.toString(), (x, y) -> new Data<>(y.getData(), y.getVersion() + 1));
+                activeTxns.compute(txId.toString(), (x, y) -> new Data<>(data.getData(), y.getVersion() + 1));
```

Comment:
This is not conditional update, it is simple update operation that changes the value of a key and increments key's version by 1.